### ISSUE TITLE
[Customer Query] Show the activity for which media is uploaded through the activity page with rtMedia Updates filter

### ISSUE THREAD

**Issue by kjnanda** [rtMediaWP/rtMedia#1423]
### Issue Description

As per the current scenario, `rtMedia Updates` filter on the activity page shows those media files for which activities are generated for media uploaded from `/media` page. 

We can use this filter to show those activities for which media files are uploaded through the activity page.

----

### Attachment

Please refer to this visual for expected behavior- https://www.loom.com/share/62d19a72db324eae8f5faff30c594345

----

### Test Environment Used

```
WordPress - V5.2.2
BuddyPress - V4.4.0
rtMedia - V4.5.8
```

**Comment by dishitpala** [comment]
@chandrapatel 
In the database, `activity_update` is not updating to `rtmedia_update` if there are media attached in the activity
And query which is fired when filters are changed to `rtMedia Updates` is
```
"SELECT DISTINCT a.*, u.user_email, u.user_nicename, u.user_login, u.display_name ,ra.privacy   FROM wp_bp_activity a LEFT JOIN wp_users u ON a.user_id = u.ID LEFT JOIN wp_rt_rtm_activity ra ON ( a.id = ra.activity_id and ra.blog_id = 1 )  WHERE  
( 
	( 
		a.user_id = '1'
	)

)
AND a.type IN ( 'rtmedia_update' ) AND a.is_spam = 0 AND a.type NOT IN ('last_activity') ORDER BY a.date_recorded DESC LIMIT 0, 20"
```
Because of this, the above issue https://www.loom.com/share/62d19a72db324eae8f5faff30c594345 is happening

Need some suggestion here
- Is it good to change the `activite_update` default type to `rtmedia_update`? 
- we have also set an action for `rtmedia_update`: `bp_activity_set_action`
- In many places there are conditional statements used like: `if ( 'rtmedia_update' === $activity->type )` but we have no value like `rtmedia_update` in database.

CC @kjnanda

**Comment by chandrapatel** [comment]
>Is it good to change the activite_update default type to rtmedia_update?

Nope. What if user just post any message without media. So may be set `rtmedia_update` activity type if any activity has media. So check accordingly.

**Comment by chandrapatel** [comment]
- When and how `rtMedia Updates` filter have been implemented?
- Was it working before?
- Is our code conflict with latest version of BuddyPress?

By above points, try to debug the issue and fix it.

**Comment by dishitpala** [comment]
@chandrapatel @kjnanda 

`rtMedia Update` Filter was introduced in buddypress-media version 4.2
- At that time it was working fine
- After applying [this](https://github.com/rtMediaWP/rtMedia/commit/a7637447c0e2d19e7009ba4a7200703ccf9f2ad3#diff-afb4ce4a3a6eaf0549c82067280b18e9L269-L278) fix maybe it was still working
- But later on, above [fix](https://github.com/rtMediaWP/rtMedia/commit/98bcf1717ee418cb271f9b8f28f74341fb7b4eeb#diff-afb4ce4a3a6eaf0549c82067280b18e9) was also removed because of some duplication issue with an activity if `Create activity for media comments` functionality is enabled

So here we only want to update an activity type to `rtmedia_update` if an activity content is having a media and activity type is `activity_update` 

CC @thrijith

**Comment by chandrapatel** [comment]
@dishitpala The code you have added back which was removed earlier due to some issues. So test thoroughly.

cc: @kjnanda

**Comment by kjnanda** [comment]
@dishitpala The added fix looks good to me.

**Comment by pooja-muchandikar** [comment]
@dishitpala @kjnanda 

I found issue with the fix added. In Profile activity page if user selects filter there is no response, user needs to explicitly refresh the page to check the filter results. This is happening for BuddyPress Legacy template only. With Nouveau template it is working fine. 

Checked with group activity page as well and issue is there with legacy template. 

```
Test Environment
rtMedia: 4.5.9 pre-release branch 
BuddyPress: 5.1.0  Nouveau and Legacy template
WordPress: 5.3.1
```

Refer to the visual -> https://drive.google.com/file/d/1YZLTzhT1_ip6YJwk11VXTmvdTnoPvOST/view
